drift/lib/src/runtime/types/mapping.dart

```diff
@@ -134,11 +134,7 @@ class SqlTypes {
     // ignore: unnecessary_cast
     switch (type as DriftSqlType<Object>) {
       case DriftSqlType.bool:
-        if (_dialect == SqlDialect.sqlite) {
-          return (sqlValue != 0) as T;
-        } else {
-          return sqlValue as T;
-        }
+        return (sqlValue != 0) as T;
```

Comment:
In postgres, we store Dart bools with the native boolean type. So when selecting a boolean column, `sqlValue` would be a boolean here. That's where the original cast comes from.

I think this is correct for some comparisons where we seem to be getting an integer back from the database, but we should still make sure to interpret `false` as `false`.

Maybe `(sqlValue != 0 && sqlValue != false) as T;`?

Reply:
Yes you are correct. For whatever reason i can not reporduce my own issue here.
So i am not really sure what happend here but i am 100% sure that yesterday the sqlvalue was 0 instead of false. Now today the sqlvalue is false and can be casted as bool. 

Anyway, I will change the return to your suggestion because that should cover all relevant cases. 👍